docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,192 @@
+
+.Second stage, run the `start` command to set any configuration option when starting the server
+<@kc.start parameters="--db-url-host=keycloak-postgres --db-username=keycloak --db-password=change_me --hostname mykeycloak.acme.com"/>
+
+At this stage, you are free to set any value you want to any of the configuration options.
+
+== Configuring the server for an optimal startup time
+
+In addition to the optimizations performed when you run the `build` command, you might want to avoid using CLI options when running the
+`start` command in favor of using environment variables or configuration options from the `conf/keycloak.conf` file.
+
+.Set any build option
+<@kc.build parameters="--db=postgres"/>
+
+.Set any configuration option to `conf/keycloak.conf`
+```
+db-url-host=keycloak-postgres
+db-username=keycloak
+db-password=change_me
+hostname mykeycloak.acme.com
+```
+
+.Start the server
+<@kc.start/>
+
+By following that approach, the server is going to skip some steps at startup and start faster.
+
+== Using the `auto-build` option when starting the server
+
+The `auto-build` option allows you to perform the two configuration stages using a single command. Under certain circumstances, you might want to sacrifice the server startup time and update the values of build options when starting the server.
+
+For that, you can start the server as follows:
+
+.Using the `auto-build` option
+<@kc.start parameters="--auto-build --db postgres --db-url-host keycloak-postgres --db-username keycloak --db-password change_me --hostname mykeycloak.acme.com"/>
+
+By using this option, the server is going to calculate the build options that have changed and automatically runs the `build` command, if necessary, before starting the server.
+
+== Configuring the server using configuration files
+
+By default, the server is going to always fetch any configuration option you set from the `conf/keycloak.conf` file. When you are using a fresh distribution,
+this file holds only the recommended settings for running in production, which are initially commented out.
+
+You can also specify a different configuration file by using the `[-cf|--config-file] option as follows:
+
+.Running the `build` command using a custom configuration file
+<@kc.build rootParameters="-cf myconfig.conf"/>
+
+.Running the `start` command using a custom configuration file
+<@kc.start rootParameters="-cf myconfig.conf"/>
+
+Changes to any *build option* defined in the `keycloak.conf` file that is targeted for the `build` command are ignored
+if the value differs from the value used to previously run the `build` command. In this case, make sure you run the `build` command again so that
+any build option is updated accordingly.
+
+== Using placeholders
+
+You are able to use placeholders to dynamically change the value of configuration options when starting the server.
+
+For that purpose, you use the following format when defining the value of an option:
+
+```
+db.username=${r"${my.db.user}"}
```

Comment:
I would have expected it to be something like:
```
db-username=${MY_ENV_VALUE}
```

Reply:
Yeah, accepted your suggestion.